src/libraries/System.Runtime.InteropServices/gen/ComInterfaceGenerator/GeneratedComInterfaceAttributeData.cs

```diff
@@ -70,6 +71,17 @@ public static GeneratedComInterfaceCompilationData GetDataFromAttribute(Attribut
                     Options = (ComInterfaceOptions)options.Value
                 };
             }
+            if (args.TryGetValue(nameof(ExceptionToUnmanagedMarshaller), out TypedConstant exceptionToUnmanagedMarshaller))
+            {
+                if (exceptionToUnmanagedMarshaller.Value is not INamedTypeSymbol)
+                {
+                    return null;
```

Comment:
We generally don't use `null` for logical flow. This would mean the return type should be updated. I don't think we should be failing in this case as it will silent and difficult to reason about. A diagnostic would be more appropriate.

Reply:
This was based on [`InteropAttributeDataExtensions.WithValuesFromNamedArguments`](https://github.com/dotnet/runtime/blob/07f3e33da71e52fce0627aa8645fc9787b954672/src/libraries/System.Runtime.InteropServices/gen/Microsoft.Interop.SourceGeneration/InteropAttributeData.cs#L82-L90), which returns `null` if an invalid `StringMarshalling` type is passed. I'm not sure how this is handled particularly by `GeneratedComInterfaceAttributeData`.

Reply:
I think this is another one of the cases where we use `null` as our "unable to parse" sentinel (and in the caller we check for null and emit a diagnostic and replace with a sentinel). Can you confirm?

Reply:
Unfortunately, it looks like there's no regards to null-checking here. Even today if you pass in an invalid type for `StringMarshallingCustomType`, i.e.
```cs
[GeneratedComInterface(StringMarshalling = StringMarshalling.Custom, StringMarshallingCustomType = typeof(string*))]
public partial interface Foo;
```
the generator fails with a generic error message from Roslyn: 
> Generator 'ComInterfaceGenerator' failed to generate source. It will not contribute to the output and compilation errors may occur as a result. Exception was of type 'NullReferenceException' with message 'Object reference not set to an instance of an object.'

Reply:
This is done.